src/frontend/screens/Library/components/SteamInstall/SteamInstallDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { Button } from '@hyperplay/ui'
+import { Dialog } from 'frontend/components/UI/Dialog'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSpinner } from '@fortawesome/free-solid-svg-icons'
+
+interface SteamInstallDialogProps {
+  isInstalling: boolean
+  setIsInstalling: (value: boolean) => void
+  onClose: () => void
+  setShowAlert: (value: 'success' | 'danger' | 'none') => void
+}
+
+const SteamInstallDialog: React.FC<SteamInstallDialogProps> = ({
+  isInstalling,
+  setIsInstalling,
+  onClose,
+  setShowAlert
+}) => {
+  const { t } = useTranslation()
+
+  const handleInstallSteam = async () => {
+    try {
+      setIsInstalling(true)
+      await window.api.installSteamWindows()
+      setShowAlert('success')
+      setTimeout(() => setShowAlert('none'), 5000)
+      setIsInstalling(false)
+      onClose()
+    } catch (error) {
+      setShowAlert('danger')
+      setTimeout(() => setShowAlert('none'), 5000)
+      setIsInstalling(false)
+    }
+  }
```

Comment:
as a rule of thumb, whenever you need to have a loading state for an action, you're better using a mutation:

```tsx
const installSteamMutation = useMutation(
    () => window.api.installSteamWindows(),
    {
      onMutate: () => {
        setIsInstalling(true);
      },
      onSuccess: () => {
        setShowAlert('success');
        setTimeout(() => setShowAlert('none'), 5000);
        setIsInstalling(false);
        onClose();
      },
      onError: () => {
        setShowAlert('danger');
        setTimeout(() => setShowAlert('none'), 5000);
        setIsInstalling(false);
      },
      onSettled: () => {
        setIsInstalling(false);
      },
    }
  );

  const handleInstallSteam = () => {
    installSteamMutation.mutate();
  };
```

```tsx
<button 
    onClick={handleInstallSteam} 
    disabled={installSteamMutation.isLoading}
>
    {installSteamMutation.isLoading ? 'Installing...' : 'Install Steam'}
</button>
```

Reply:
I think I would need more changes than that though:
```Type '() => Promise<void>' has no properties in common with type 'UseMutationOptions<unknown, Error, void, unknown>'.ts(2559)```

the only improvement is that I won't need to pass the setState?  because it seems more code and the actual code is working fine afaik.